packages/graphback-runtime-mongodb/src/MongoDBDataProvider.ts

```diff
@@ -165,6 +178,60 @@ export class MongoDBDataProvider<Type = any, GraphbackContext = any> implements
     return document;
   }
 
+  protected getIndexFields(baseType: GraphQLObjectType): IndexSpecification[] {
+    const res: IndexSpecification[] = [];
+    const fields = baseType.getFields();
+    Object.keys(fields).forEach((k: string) => {
+      const field = fields[k];
+
+      // Add custom Index if found e.g. @db(index: 1)
+      const customIndex = this.getCustomIndex(field);
+      if (customIndex !== undefined) {
+        res.push(customIndex);
+
+        // To prevent adding mutliple index on same field
+        return;
+      }
+      
+      // Add Index on relation fields
+      const relationIndex = this.getRelationIndex(field);
+      if (relationIndex !== undefined) {
+        res.push(relationIndex);
+      }
+
+    })
+    
+    return res;
+  }
+
+  private getCustomIndex(field: GraphQLField<any, any>): IndexSpecification {
+    const otherAnnotations = parseMetadata('db', field.description) as any;
+      if (otherAnnotations?.index) {
+        return {
+          key: {
+            [field.name]: 1
+          },
+          name: `graphback_custom_index_${field.name}_on_${this.collectionName}`
```

Comment:
It'd be awesome if we could be able to provide index name too. 

e.g via a `db.index: 'index-name-here'`

/cc @craicoverflow @wtrocki

Reply:
+1 on that @machi1990 

It could go one step further and parse the index annotation directly as an `IndexSpecification` interface, falling back on defaults otherwise?

```
"""
@index(
  name: 'my_index',
  unique: true,
  expiresAfterSeconds: 3600,
  collationDocument: {
    locale: 'en_IE'
  }
)
"""
```

Reply:
Sounds cool! How about:
```
@index(name: 'index123')
```
With this, we could even use `@index` and it would use a default index name

Reply:
+1 on all your  suggestions. Good stuffs.

Reply:
> With this, we could even use @index and it would use a default index name

Exactly. Only `key` is required in `IndexSpecification`.

>  If unspecified, MongoDB generates an index name by concatenating the names of the indexed fields and the sort order.